doc/device_client.md

```diff
@@ -0,0 +1,32 @@
+# Azure IoT Device Client Library
+
+The Azure IoT Device Client Library provides an efficient and well tested API for connecting and using devices with the Azure IoT Hub.
+
+## client.setOptions(options, [callback])
+
+Used directly after creating the Device Client to set options on the client. These should not be dynamically altered during runtime of the device client.
+
+### Available Options
```

Comment:
Will these options also show up somewhere in API reference docs?

Reply:
...no...

Reply:
ok, should they? seems to me like it would be helpful but if that's breaking API reference doc precedent that's fine

Reply:
wait sorry, these options do show up in the API somewhere, but not in this form. I pulled most of the definitions for the options from the docstrings.